src/main/java/com/example/onlinebookstore/service/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,123 @@
+package com.example.onlinebookstore.service.impl;
+
+import com.example.onlinebookstore.dto.order.OrderItemResponseDto;
+import com.example.onlinebookstore.dto.order.OrderRequestDto;
+import com.example.onlinebookstore.dto.order.OrderResponseDto;
+import com.example.onlinebookstore.dto.order.OrderStatusRequestDto;
+import com.example.onlinebookstore.exceptions.EntityNotFoundException;
+import com.example.onlinebookstore.mapper.OrderItemMapper;
+import com.example.onlinebookstore.mapper.OrderMapper;
+import com.example.onlinebookstore.model.CartItem;
+import com.example.onlinebookstore.model.Order;
+import com.example.onlinebookstore.model.OrderItem;
+import com.example.onlinebookstore.model.ShoppingCart;
+import com.example.onlinebookstore.model.User;
+import com.example.onlinebookstore.repository.OrderItemRepository;
+import com.example.onlinebookstore.repository.OrderRepository;
+import com.example.onlinebookstore.repository.ShoppingCartRepository;
+import com.example.onlinebookstore.repository.UserRepository;
+import com.example.onlinebookstore.service.OrderService;
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@RequiredArgsConstructor
+public class OrderServiceImpl implements OrderService {
+    private final UserRepository userRepository;
+    private final OrderMapper orderMapper;
+    private final OrderRepository orderRepository;
+    private final OrderItemRepository orderItemRepository;
+    private final OrderItemMapper orderItemMapper;
+    private final ShoppingCartRepository shoppingCartRepository;
+
+    @Transactional
+    @Override
+    public OrderResponseDto makeOrder(OrderRequestDto orderRequestDto, Long userId) {
+        User user = userRepository.findById(userId)
+                .orElseThrow(() -> new EntityNotFoundException("Can't find user by id: " + userId));
+        ShoppingCart shoppingCart = shoppingCartRepository.findShoppingCartByUserId(userId)
+                .orElseThrow(() -> new EntityNotFoundException(
+                        "Can't find shopping cart by this userId: " + userId));
+        Order order = createOrder(shoppingCart.getCartItems());
+
+        order.setUser(user);
+        order.setOrderDate(LocalDateTime.now());
+        order.setStatus(Order.Status.PENDING);
+        order.setShippingAddress(orderRequestDto.shippingAddress());
+        order.setTotal(calculateTotal(order.getOrderItems()));
+
+        shoppingCart.clearCartItems();
+        orderRepository.save(order);
+
+        return orderMapper.toResponseOrder(order);
+    }
+
+    @Transactional(readOnly = true)
```

Comment:
minor

you can put this annotation on class instead of duplicating it for read methods

and then simply override it for write methods